Added `defmulti` to list of functions that get annotated in defs-db

callendorph opened the pull request:
This commit causes the definitions database to create `SrcDefMulti` kinds of definitions. This means that the `defmulti` declarations for a particular `deftype` will be present in a generated definitions database.

This should help when defining the documentation for a `deftype` interface as the documentation generator will be able to determine what interface defmulti's will apply.

This only seems to be applicable to the definitions created from the source files. Correct me if I'm wrong, but it seems like the `*.pkg` definitions already extract the `MultiRec` which seems to be `defmulti`.

I've tested this locally with the documentation generator and I see the following definitions created:

```
  Definition ESeries defined at /Users/callendorph/src/docgen/test/data/basic/src/deftype.stanza:102.15:
    kind: SrcDefType
    visibility: Public
    annotation?: false
    documentation?:

      E-series Preferred Numbers Base Type
      @see https://en.wikipedia.org/wiki/E_series_of_preferred_numbers

  Definition name defined at /Users/callendorph/src/docgen/test/data/basic/src/deftype.stanza:108.16:
    kind: SrcDefMulti
    visibility: Public
    annotation?: (self:ESeries) -> String
    documentation?:

      Get the Name fo the Series
      @return Series Name in form "E?" where ? is 12, 24, 48, etc

  Definition elements defined at /Users/callendorph/src/docgen/test/data/basic/src/deftype.stanza:113.16:
    kind: SrcDefMulti
    visibility: Public
    annotation?: (self:ESeries) -> Int
    documentation?:

      Get the total number of elements in this series.
      @return Total Number of Series Elements

```

Which matches with the input stanza:

```
doc: \<DOC>
E-series Preferred Numbers Base Type
@see https://en.wikipedia.org/wiki/E_series_of_preferred_numbers
<DOC>
public deftype ESeries

doc: \<DOC>
Get the Name fo the Series
@return Series Name in form "E?" where ? is 12, 24, 48, etc
<DOC>
public defmulti name (self:ESeries) -> String
doc: \<DOC>
Get the total number of elements in this series.
@return Total Number of Series Elements
<DOC>
public defmulti elements (self:ESeries) -> Int
```

CuppoJava commented on compiler/stz-config.stanza:
```diff
@@ -36,6 +36,11 @@ defsyntax stanza-config :
   fail-if long! = () : PE(closest-info(), "Expected a long here.")
 
+  defproduction bool! : True|False
+  defrule bool! = (?v:true) : true
```
defrule bool! = (?v) when unwrap-token(v) is True|False : unwrap-token(v)

callendorph committed:
Added `defmulti` to list of functions that get annotated in defs-db

This commit causes the definitions database to create `SrcDefMulti`
kinds of definitions. This means that the `defmulti` declarations for a
particular `deftype` will be present in a generated definitions database.

This should help when defining the documentation for a `deftype` interface
as the documentation generator will be able to determine what interface
defmulti's will apply.

callendorph committed:
Modifies the definition annotation printing to match definition syntax

Previously the definitions that were created by printing the
DefnAnnotation did not have a space between the type parameter
and the argument list. For example, you might get these outputs

```
(a:Int) -> String
<T?>(a:T?) -> T
```

For the first example, there isn't anything inherently wrong.
Bu the second I believe you not compile because there is no space between
the `<>` and the `()`

This adds a prefix space in both cases. This makes it easier to apply
this directly to a function defn signature:

```
defn some-func (a:Int) -> String
defn another<T?> (a:T?) -> T
```
